### PR TITLE
Volume: Introduce ray shuffling to limit banding artifacts

### PR DESCRIPTION
**Pull request by skalarproduktraum**
This PR adds shuffling to cast rays in order to reduce banding artifacts.

### CODE REVIEW

File: src/test/kotlin/graphics/scenery/tests/examples/volumes/FlybrainOutOfCoreExample.kt
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/volumes/FlybrainOutOfCoreExample.kt:1]: getDemoFilesPath() + "/volumes/flybrain.zip" is incorrect

File: src/test/kotlin/graphics/scenery/tests/examples/volumes/OutOfCoreRAIExample.kt [hunk 1]
```diff
@@ -43,7 +43,7 @@ class OutOfCoreRAIExample: SceneryBase("Out-of-core RAI Rendering example", 1280
         scene.addChild(cam)
 
         // download example TIFF stack, T1 head example from NIH
-        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val imp: ImagePlus = IJ.openImage(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/volumes/OutOfCoreRAIExample.kt:46]: This .zip is wrong

File: src/test/kotlin/graphics/scenery/tests/examples/volumes/RAIExample.kt [hunk 2]
```diff
@@ -35,10 +35,11 @@ class RAIExample: SceneryBase("RAI Rendering example", 1280, 720) {
             scene.addChild(this)
         }
 
-        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val imp: ImagePlus = IJ.openImage(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/volumes/RAIExample.kt:38]: This .zip is wrong

File: src/test/kotlin/graphics/scenery/tests/examples/volumes/NetworkVolumeExample.kt [hunk 3]
```diff
@@ -55,7 +55,7 @@ class NetworkVolumeExample : SceneryBase("SpimData Rendering example", 1280, 720
             Volume.VolumeFileSource.VolumeType.TIFF
         )
 
-        val online = IJVolumeInitializer("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val online = IJVolumeInitializer(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/volumes/NetworkVolumeExample.kt:58]: Also wrong path

File: src/test/kotlin/graphics/scenery/tests/examples/volumes/OrthoViewExample.kt [hunk 4]
```diff
@@ -50,11 +50,12 @@ class OrthoViewExample : SceneryBase("Ortho View example", 1280, 720) {
         origin.material().diffuse = Vector3f(0.8f, 0.0f, 0.0f)
         scene.addChild(origin)
 
-        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val imp: ImagePlus = IJ.openImage(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/volumes/OrthoViewExample.kt:53]: Also wrong path

File: src/test/kotlin/graphics/scenery/tests/examples/compute/PersistentTextureRequestsExample.kt [hunk 5]
```diff
@@ -58,7 +58,7 @@ class PersistentTextureRequestsExample : SceneryBase("PersistentTextureRequestsE
 
         hub.add(volumeManager)
 
-        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val imp: ImagePlus = IJ.openImage(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/compute/PersistentTextureRequestsExample.kt:61]: Also wrong path

File: src/test/kotlin/graphics/scenery/tests/examples/compute/CustomVolumeManagerExample.kt [hunk 6]
```diff
@@ -49,7 +49,7 @@ class CustomVolumeManagerExample : SceneryBase("CustomVolumeManagerExample") {
 
         hub.add(volumeManager)
 
-        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val imp: ImagePlus = IJ.openImage(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/compute/CustomVolumeManagerExample.kt:52]: Also wrong path

File: src/test/kotlin/graphics/scenery/tests/examples/advanced/VRVolumeCroppingExample.kt [hunk 7]
```diff
@@ -60,7 +60,7 @@ class VRVolumeCroppingExample : SceneryBase(VRVolumeCroppingExample::class.java.
 
         scene.addChild(cam)
 
-        val imp: ImagePlus = IJ.openImage("https://imagej.nih.gov/ij/images/t1-head.zip")
+        val imp: ImagePlus = IJ.openImage(getDemoFilesPath() + "/volumes/t1-head.zip")
```
Comment (kephale) [inline src/test/kotlin/graphics/scenery/tests/examples/advanced/VRVolumeCroppingExample.kt:63]: Also wrong path

**Review by kephale — CHANGES_REQUESTED** [review]
Tested on BDVExample and ProceduralVolumeExample and the viz looks great.

Many examples have the wrong path for the volumes

File: src/test/kotlin/graphics/scenery/tests/examples/volumes/OutOfCoreRAIExample.kt (hunk 1 above)
Comment (skalarproduktraum) [reply src/test/kotlin/graphics/scenery/tests/examples/volumes/OutOfCoreRAIExample.kt:46]: You might have an old models.zip file. The current one has both the .zip and the .xml/.h5 👍

**Comment by skalarproduktraum** [comment]
@kephale the paths are not wrong, I have updated the demo models zip file yesterday, with `flybrain.zip` and `t1-head.zip`:
image

**Review by kephale — APPROVED** [review]

**Pull request merged**